activiti-core/activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/tasks/TaskRuntimeTaskAssigneeTest.java

```diff
@@ -228,5 +229,73 @@ public void createStandaloneTaskForUsersAndAdminReassignUser() {
                                               .build());
     }
 
+    @Test
+    public void userCanReassignClaimedTaskToCandidateUsers() {
+
+        String taskId = createTask("garth");
+        //assign the new task
+        userAssignTask(taskId,"garth", "dean");
+        //reassign task to candidate
+        userAssignTask(taskId,"dean", "garth");
+
+        test_shouldReturnIllegalStateExceptionWhenUserIsNotAssigneeOrOwner(taskId,"dean", "garth");
+
+        cleanUp(taskId);
```

Comment:
No need to clean up here. It's already done by [the @AfterEach method](https://github.com/Activiti/Activiti/pull/3440/files#diff-9a5a1effbdd6cbccc0d0cca73867d471R51)